### PR TITLE
Use channel to propagate the stop info of sandbox

### PR DESCRIPTION
**Pull request by miaoyq**
Address this [comment ](https://github.com/containerd/cri-containerd/pull/571#discussion_r163731014) in #571
Related to #569 

Signed-off-by: Yanqiang Miao <miao.yanqiang@zte.com.cn>

### CODE REVIEW

File: pkg/server/sandbox_stop.go [hunk 1]
```diff
@@ -125,25 +121,16 @@ func (c *criContainerdService) stopSandboxContainer(ctx context.Context, sandbox
 	return c.waitSandboxStop(ctx, sandbox, killContainerTimeout)
 }
 
-// waitSandboxStop polls sandbox state until timeout exceeds or sandbox is stopped.
+// waitSandboxStop waits for sandbox to be stopped until timeout exceeds or centext is cancelled.
```
Comment (Random-Liu) [inline pkg/server/sandbox_stop.go]: s/centext/context

File: pkg/store/sandbox/sandbox.go [hunk 2]
```diff
@@ -36,15 +37,22 @@ type Sandbox struct {
 	Container containerd.Container
 	// CNI network namespace client
 	NetNS *NetNS
+	// StopCh is used to propagate the stop information of the sandbox.
+	container.StopCh
```
Comment (Random-Liu) [inline pkg/store/sandbox/sandbox.go]: Let's move stopCh to a separate file under `pkg/store` for now.
Don't want sandbox store and container store import each other.

File: pkg/server/sandbox_stop.go (hunk 1 above)
Comment (miaoyq) [reply pkg/server/sandbox_stop.go]: Done.

File: pkg/store/sandbox/sandbox.go (hunk 2 above)
Comment (miaoyq) [reply pkg/store/sandbox/sandbox.go]: Done.

**Pull request merged**